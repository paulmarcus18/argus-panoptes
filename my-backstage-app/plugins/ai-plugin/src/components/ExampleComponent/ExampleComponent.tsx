import React, { useEffect, useState } from 'react';
import { Card, CardContent, Typography, CircularProgress } from '@mui/material';
import { GoogleGenAI } from '@google/genai';  // Import the GoogleGenAI package

// Initialize GoogleGenAI with your API key
const ai = new GoogleGenAI({ apiKey: "AIzaSyAGvFE_8fGwGtBNzSL8_x6lbxaj24xUOBE" });

const commitMessages = [
  'Fix bug in login logic when email is missing',
  'Refactor signup flow to use new auth API',
  'Update dependencies and remove deprecated packages',
  'Improve error logging in user service',
  'Add unit tests for checkout module',
];

// Build the prompt to generate the summary
const prompt = `
You are a release note generator. Given the following GitHub commit messages, summarize the changes in a clear, professional paragraph:

${commitMessages.map(msg => `- ${msg}`).join('\n')}
`;

export const ExampleComponent = () => {
  const [summary, setSummary] = useState('');
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const generateSummary = async () => {
      try {
        // Make a request to the Gemini API via the GoogleGenAI package
        const response = await ai.models.generateContent({
          model: 'gemini-2.0-flash',  // Specify the correct model name here
          contents: prompt,           // Pass the prompt to the API
        });

        // Extract the text from the API response
        const result = response.text || 'No summary returned.';
        setSummary(result);  // Set the result into state
      } catch (error) {
        console.error('Error generating summary:', error);
        setSummary('‚ùå Error generating summary.');
      } finally {
        setLoading(false);  // Turn off loading state
      }
    };

    // Trigger the summary generation when the component mounts
    generateSummary();
  }, []);  // Empty dependency array ensures the effect runs only once

  return (
    <Card>
      <CardContent>
        <Typography variant="h6">AI-Generated Summary</Typography>
        {loading ? <CircularProgress /> : <Typography>{summary}</Typography>}
      </CardContent>
    </Card>
  );
};
