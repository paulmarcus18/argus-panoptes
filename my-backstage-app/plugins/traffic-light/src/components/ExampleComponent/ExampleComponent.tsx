import { Typography, Grid, Box } from '@material-ui/core';
import {
  InfoCard,
  Header,
  Page,
  Content,
  ContentHeader,
  HeaderLabel,
  SupportButton,
} from '@backstage/core-components';
import { ExampleFetchComponent } from '../ExampleFetchComponent';

export const ExampleComponent = () => (
  <Page themeId="tool">
    <Header title="Welcome to traffic-light!" subtitle="Optional subtitle">
      <HeaderLabel label="Owner" value="Team X" />
      <HeaderLabel label="Lifecycle" value="Alpha" />
    </Header>
    <Content>
      <ContentHeader title="Plugin title">
        <SupportButton>A description of your plugin goes here.</SupportButton>
      </ContentHeader>
      <Grid container spacing={3} direction="column">
        <Grid item>
          <InfoCard title="Information card">
            <Typography variant="body1">
              All content should be wrapped in a card like this.
            </Typography>
          </InfoCard>
        </Grid>
        <Grid item>
          <ExampleFetchComponent />
        </Grid>
      </Grid>
      <Box display="flex" justifyContent="space-around" alignItems="center" mt={4}>
      <div style={{
        width: 100,
        height: 100,
        borderRadius: '50%',
        backgroundColor: 'red',
      }} />
      <div style={{
        width: 100,
        height: 100,
        borderRadius: '50%',
        backgroundColor: 'green',
      }} />
    </Box>
    </Content>
  </Page>
);
