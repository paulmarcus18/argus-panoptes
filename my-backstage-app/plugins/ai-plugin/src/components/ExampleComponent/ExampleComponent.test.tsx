import React from 'react';
import { render, screen } from '@testing-library/react';
import { ExampleComponent } from './ExampleComponent';

// Mock the dependent components to isolate the test
jest.mock('@backstage/core-components', () => ({
  InfoCard: ({ title, children }: { title: string; children: React.ReactNode }) => (
    <div data-testid="info-card" data-title={title}>
      {children}
    </div>
  ),
  Header: ({ title, subtitle, children }: { title: string; subtitle?: string; children: React.ReactNode }) => (
    <div data-testid="header" data-title={title} data-subtitle={subtitle}>
      {children}
    </div>
  ),
  HeaderLabel: ({ label, value }: { label: string; value: string }) => (
    <div data-testid="header-label" data-label={label} data-value={value} />
  ),
  Page: ({ themeId, children }: { themeId: string; children: React.ReactNode }) => (
    <div data-testid="page" data-theme-id={themeId}>
      {children}
    </div>
  ),
  Content: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="content">{children}</div>
  ),
  ContentHeader: ({ title, children }: { title: string; children: React.ReactNode }) => (
    <div data-testid="content-header" data-title={title}>
      {children}
    </div>
  ),
  SupportButton: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="support-button">{children}</div>
  ),
}));

// Mock the ExampleFetchComponent
jest.mock('../ExampleFetchComponent', () => ({
  ExampleFetchComponent: () => <div data-testid="example-fetch-component" />,
}));

// Mock Material-UI components
jest.mock('@material-ui/core', () => ({
  Typography: ({ variant, children }: { variant: string; children: React.ReactNode }) => (
    <div data-testid="typography" data-variant={variant}>
      {children}
    </div>
  ),
  Grid: ({ 
    container, 
    item, 
    spacing, 
    direction, 
    children 
  }: { 
    container?: boolean; 
    item?: boolean;
    spacing?: number;
    direction?: string;
    children: React.ReactNode 
  }) => (
    <div 
      data-testid="grid" 
      data-container={container ? 'true' : 'false'} 
      data-item={item ? 'true' : 'false'}
      data-spacing={spacing}
      data-direction={direction}
    >
      {children}
    </div>
  ),
}));

describe('ExampleComponent', () => {
  it('should render the component structure correctly', () => {
    render(<ExampleComponent />);
    
    // Test the Page component
    const page = screen.getByTestId('page');
    expect(page).toHaveAttribute('data-theme-id', 'tool');
    
    // Test the Header component
    const header = screen.getByTestId('header');
    expect(header).toHaveAttribute('data-title', 'Welcome to ai-plugin!');
    expect(header).toHaveAttribute('data-subtitle', 'Optional subtitle');
    
    // Test HeaderLabels
    const headerLabels = screen.getAllByTestId('header-label');
    expect(headerLabels).toHaveLength(2);
    expect(headerLabels[0]).toHaveAttribute('data-label', 'Owner');
    expect(headerLabels[0]).toHaveAttribute('data-value', 'Team X');
    expect(headerLabels[1]).toHaveAttribute('data-label', 'Lifecycle');
    expect(headerLabels[1]).toHaveAttribute('data-value', 'Alpha');
    
    // Test ContentHeader
    const contentHeader = screen.getByTestId('content-header');
    expect(contentHeader).toHaveAttribute('data-title', 'Plugin title');
    
    // Test SupportButton
    const supportButton = screen.getByTestId('support-button');
    expect(supportButton).toHaveTextContent('A description of your plugin goes here.');
    
    // Test InfoCard
    const infoCard = screen.getByTestId('info-card');
    expect(infoCard).toHaveAttribute('data-title', 'Information card');
    
    // Test Typography
    const typography = screen.getByTestId('typography');
    expect(typography).toHaveAttribute('data-variant', 'body1');
    expect(typography).toHaveTextContent('All content should be wrapped in a card like this.');
    
    // Test that ExampleFetchComponent is rendered
    expect(screen.getByTestId('example-fetch-component')).toBeInTheDocument();
  });

  it('should render the grid structure correctly', () => {
    render(<ExampleComponent />);
    
    // Get all grid elements
    const grids = screen.getAllByTestId('grid');
    
    // Check the outer container Grid
    expect(grids[0]).toHaveAttribute('data-container', 'true');
    expect(grids[0]).toHaveAttribute('data-spacing', '3');
    expect(grids[0]).toHaveAttribute('data-direction', 'column');
    
    // Check there are two Grid items
    expect(grids[1]).toHaveAttribute('data-item', 'true');
    expect(grids[2]).toHaveAttribute('data-item', 'true');
  });

  it('matches snapshot', () => {
    const { container } = render(<ExampleComponent />);
    expect(container).toMatchSnapshot();
  });
});