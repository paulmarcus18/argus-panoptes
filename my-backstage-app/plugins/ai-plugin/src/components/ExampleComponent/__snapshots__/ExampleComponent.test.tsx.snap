// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent matches snapshot 1`] = `
<div>
  <div
    data-testid="page"
    data-theme-id="tool"
  >
    <div
      data-subtitle="Optional subtitle"
      data-testid="header"
      data-title="Welcome to ai-plugin!"
    >
      <div
        data-label="Owner"
        data-testid="header-label"
        data-value="Team X"
      />
      <div
        data-label="Lifecycle"
        data-testid="header-label"
        data-value="Alpha"
      />
    </div>
    <div
      data-testid="content"
    >
      <div
        data-testid="content-header"
        data-title="Plugin title"
      >
        <div
          data-testid="support-button"
        >
          A description of your plugin goes here.
        </div>
      </div>
      <div
        data-container="true"
        data-direction="column"
        data-item="false"
        data-spacing="3"
        data-testid="grid"
      >
        <div
          data-container="false"
          data-item="true"
          data-testid="grid"
        >
          <div
            data-testid="info-card"
            data-title="Information card"
          >
            <div
              data-testid="typography"
              data-variant="body1"
            >
              All content should be wrapped in a card like this.
            </div>
          </div>
        </div>
        <div
          data-container="false"
          data-item="true"
          data-testid="grid"
        >
          <div
            data-testid="example-fetch-component"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
